src/exportJarFileCommand.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { EOL, platform } from "os";
+import { commands, Uri, window } from "vscode";
+import { sendOperationError } from "vscode-extension-telemetry-wrapper";
+import { buildWorkspace } from "./build";
+import { IStep } from "./exportJarSteps/IStep";
+import { StepGenerateJar } from "./exportJarSteps/StepGenerateJar";
+import { StepResolveMainMethod } from "./exportJarSteps/StepResolveMainMethod";
+import { StepResolveWorkspace } from "./exportJarSteps/StepResolveWorkspace";
+import { isStandardServerReady } from "./extension";
+import { INodeData } from "./java/nodeData";
+
+let isExportingJar: boolean = false;
+
+export class StepMetadata {
+    public entry?: INodeData;
+    public workspaceUri?: Uri;
+    public isPickedWorkspace: boolean;
+    public projectList?: INodeData[];
+    public selectedMainMethod?: string;
+    public outputPath?: string;
+    public elements: string[];
+}
+
+export namespace steps {
+    export const stepResolveWorkspace: StepResolveWorkspace = new StepResolveWorkspace();
+    export const stepResolveMainMethod: StepResolveMainMethod = new StepResolveMainMethod();
+    export const stepGenerateJar: StepGenerateJar = new StepGenerateJar();
+    export let currentStep: number = 0;
+    export const stepsList: IStep[] = [stepResolveWorkspace, stepResolveMainMethod, stepGenerateJar];
```

Comment:
Why not use a `Map<ExportJatStep, IExportJarStepExecutor>` to map them. Using array and step counter is hard to the scenario that needs to skip steps